components/admin_console/system_users/system_users.jsx

```diff
@@ -132,19 +136,25 @@ export default class SystemUsers extends React.Component {
 
     handleTeamChange(e) {
         const teamId = e.target.value;
-        this.loadDataForTeam(teamId);
-        this.props.actions.setSystemUsersSearch(this.props.searchTerm, teamId);
+        this.loadDataForTeam(teamId, this.props.filter);
+        this.props.actions.setSystemUsersSearch(this.props.searchTerm, teamId, this.props.filter);
+    }
+
+    handleFilterChange(e) {
+        const filter = e.target.value;
+        this.props.actions.setSystemUsersSearch(this.props.searchTerm, this.props.teamId, filter);
+        this.loadDataForTeam(this.props.teamId, filter);
```

Comment:
nit: these two lines are swapped between the function above. Maybe swap back for consistency?